x-pack/plugins/canvas/public/apps/export/routes.js

```diff
@@ -32,6 +33,7 @@ export const routes = [
           dispatch(setAssets(assets));
           dispatch(setWorkpad(workpad, { loadPages: false }));
           dispatch(setPage(pageNumber - 1));
+          dispatch(selectToplevelNodes([]));
```

Comment:
Q: Was this just an oversight in the old code, where selected elements were not cleared?

Reply:
In the old code, we cleared the selection via the middleware (I think circuitously, by clearing it in the `aero` store which then updated back `redux`) so it just used a different mechanism there. But glad you asked because:

I'm not proposing this for this PR, but adding this line (and similar ones) made me ponder if our currently fairly lean reducers should be able to do more. For example, here, we issue four dispatches. `combineReducers` is nice when state slices are truly independent, but it's not possible to implement a new version of `setPage` that would also vacate the `transient.selectedToplevelNodes` array. I'm biased toward one user action = one Redux action (sans async stuff like server response) and the Redux doc has [a pertinent note](https://redux.js.org/api/combinereducers#notes) on `combineReducers`.

Reply:
I also prefer to lean towards the 1:1 user action to redux action route.

Would it make sense to update the `transientReducer` to be watching for any of these "page changed" type actions (add, set, delete, duplicate, etc) and set the `selectedTopLevelNodes` state to empty array? Rather than having to always dispatch this selectTopLevelNodes action when any kind of page change happens?

```
if (action is changePage or deletePage or addPage)
  selectedTopLevelElements = []
```

Reply:
That's probably the better "redux" way, having the transient reducer update based on these actions. So that dispatching only `setWorkpad` or `setPage` does what you want without needing to dispatch other actions all the time.

There's a lot of other places we could clean up like that too.